### ISSUE TITLE
Test runner's command parser needs improvement

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#1238]
(I did an addition for support of french quotes in executing steps in test suite.) And now I would like to execute following godog step:
```
When executing »minishift openshift config set --patch '{"assetConfig": {"logoutURL": "http://www.minishift.io"}}'« succeeds
```
but currently the test runner has several problems with it. First is that there are whitespaces which parser in test runner just uses as a split character:
```golang
commandArr := strings.Split(command, " ")
path, _ := filepath.Abs(commandPath)
cmd := exec.Command(path, commandArr...)
```
and so the string after --patch flag is not handled as single parameter, but as 3 parameters. If I remove the whitespaces, then second problem arises and it is that there are extra quotes then. It needs to be:
```
When executing »minishift openshift config set --patch {"assetConfig":{"logoutURL":"http://www.minishift.io"}} succeeds
```
this runs fine, but it is not a command which normal terminal would consume.

One solution is to use parsing package, for example: github.com/mattn/go-shellwords. We could then parse probably like this:
```golang
commandArr, err := shellwords.Parse(command)
//commandArr then should be: ["openshift", "config", "set", "--patch", "{"assetConfig":{"logoutURL":"http://www.minishift.io"}}"]
```

WDYT?

**Comment by hferentschik** [comment]
I think we have parts of this already covered by the OcRunner in the production code - https://github.com/minishift/minishift/blob/4b9ddbf40f525d63a86eac087b1406997b1a5ce9/pkg/minishift/oc/oc.go#L39-L39

See also https://github.com/minishift/minishift/blob/51d21cd06b7753cfc0c62984bf23ffd796cc8c29/pkg/util/cmd/split_args.go#L29-L29

We should be aligning with this.

**Comment by agajdosi** [comment]
@hferentschik Thanks a lot! Going to look at it.

**Comment by gbraad** [comment]
in other words, the code should move from integration to the general
codebase... and instead re-used in the integration test. Pretty much like
what happened with the hostip code earlier

**Comment by agajdosi** [comment]
@hferentschik @gbraad `github.com/minishift/minishift/pkg/util/cmd` works like a charm, thanks for suggestions, this was fast! Will make it part of #1228.

**Comment by hferentschik** [comment]
cool

**Comment by gbraad** [comment]
Do we still need this issue report or will it be a refactor as part of #1228?

**Comment by agajdosi** [comment]
@gbraad Will be part of #1228.

**Comment by agajdosi** [comment]
Merged via pull request #1228
